packages/jupyter-ai/jupyter_ai/chat_handlers/learn.py

```diff
@@ -119,26 +133,28 @@ def _build_list_response(self):
         {dir_list}"""
         return message
 
-    async def learn_dir(self, path: str):
+    async def learn_dir(
+        self, path: str, chunk_size: int = 2000, chunk_overlap: int = 100
+    ):
         dask_client = await self.dask_client_future
         splitters = {
             ".py": PythonCodeTextSplitter(
-                chunk_size=self.chunk_size, chunk_overlap=self.chunk_overlap
+                chunk_size=chunk_size, chunk_overlap=chunk_overlap
             ),
             ".md": MarkdownTextSplitter(
-                chunk_size=self.chunk_size, chunk_overlap=self.chunk_overlap
+                chunk_size=chunk_size, chunk_overlap=chunk_overlap
             ),
             ".tex": LatexTextSplitter(
-                chunk_size=self.chunk_size, chunk_overlap=self.chunk_overlap
+                chunk_size=chunk_size, chunk_overlap=chunk_overlap
             ),
             ".ipynb": NotebookSplitter(
-                chunk_size=self.chunk_size, chunk_overlap=self.chunk_overlap
+                chunk_size=chunk_size, chunk_overlap=chunk_overlap
             ),
         }
         splitter = ExtensionSplitter(
             splitters=splitters,
             default_splitter=RecursiveCharacterTextSplitter(
-                chunk_size=self.chunk_size, chunk_overlap=self.chunk_overlap
+                chunk_size=chunk_size, chunk_overlap=chunk_overlap
             ),
         )
```

Comment:
Nit: I would prefer first defining

```py
splitter_kwargs = {
  "chunk_size": chunk_size,
  "chunk_overlap": chunk_overlap,
}
```

and then initializing each splitter with `**splitter_kwargs`.